src/mpyl/run_plan.py

```diff
@@ -1,38 +1,61 @@
 """This module contains the RunPlan class."""
-
 from dataclasses import dataclass
+from typing import Optional
 
 from .project import Project, Stage
 from .project_execution import ProjectExecution
 
 
-@dataclass(frozen=True)
+@dataclass()
 class RunPlan:
```

Comment:
Feels a bit weird to have a non frozen "dataclass" that actually has state (with the add_state and update methods). Can't we get rid of those and initialise this dataclass with the full runplan? Then we can make it frozen as well. If it's not possible then I would make this a normal class